File: learn-nodejs/test/leetcode-ts/leetcode_830.test.ts
function largeGroupPositions(s: string): number[][] {
    let arr = new Array(), sub = 0, length = s.length;
    for (let i = 0; i < length; i++) {
        if (s[sub] != s[i]) {
            if (i - sub >= 3) {
                arr.push([sub, i - 1]);
            }
            sub = i;
        }
    }
    if (length - sub >= 3) {
        arr.push([sub, length - 1]);
    }
    return arr;
};

/**
 * 830. 较大分组的位置
 */
test('leetcode_830', () => {
    console.log(largeGroupPositions("abbxxxxzzy"))
    console.log(largeGroupPositions("abc"))
    console.log(largeGroupPositions("abcdddeeeeaabbbcd"))
    console.log(largeGroupPositions("aba"))
    console.log(largeGroupPositions("aaa"))
});

File: learn-golang/test/leetcode/leetcode_830_test.go
package leetcode

import (
	"fmt"
	"testing"
)

/**
 * 830. 较大分组的位置
 */

func Test_leetcode_830(t *testing.T) {
	fmt.Println(largeGroupPositions("abbxxxxzzy"))
	fmt.Println(largeGroupPositions("abc"))
	fmt.Println(largeGroupPositions("abcdddeeeeaabbbcd"))
	fmt.Println(largeGroupPositions("aba"))
	fmt.Println(largeGroupPositions("aaa"))
}

func largeGroupPositions(s string) [][]int {
	defer timeCost()()
	arr, sub, length := make([][]int, 0), 0, len(s)
	for i := 1; i < length; i++ {
		if s[sub] != s[i] {
			if i-sub >= 3 {
				arr = append(arr, []int{sub, i - 1})
			}
			sub = i
		}
	}
	if length-sub >= 3 {
		arr = append(arr, []int{sub, length - 1})
	}
	return arr
}
